optuna/cli.py

```diff
@@ -338,6 +338,82 @@ def take_action(self, parsed_args: Namespace) -> None:
         self.logger.info("Attribute successfully written.")
 
 
+class _TrialSuggest(_BaseCommand):
+    """Suggest a parameter."""
+
+    @staticmethod
+    def _int_or_float(v: str) -> Union[int, float]:
+        try:
+            return int(v)
+        except ValueError:
+            return float(v)
+
+    def get_parser(self, prog_name: str) -> ArgumentParser:
+
+        parser = super(_TrialSuggest, self).get_parser(prog_name)
+        parser.add_argument("param_name")
+        parser.add_argument("--study-name", required=True)
+        parser.add_argument("--trial-number", required=True, type=int)
+        parser.add_argument("--distribution", choices=("float", "int", "categorical"))
+        parser.add_argument("--choices", nargs="+")
+        parser.add_argument("--high", type=self._int_or_float)
+        parser.add_argument("--low", type=self._int_or_float)
+        parser.add_argument("--step", type=self._int_or_float)
+        parser.add_argument("--log-domain", action="store_true", default=False)
```

Comment:
I think we can simplify these arguments by introducing `--distribution` as a string of a distribution class like
```
optuna suggest ... --distribution UniformDistribution(0, 1) ...
```

Reply:
Thank you for the suggestion. So, would it be nice to make `optuna trial suggest` have `--search-space` as in `optuna ask`? (if it accepts `--search-space`, we can sample multiple parameters at a time)

Reply:
Hmm... If the `optuna trial suggest` accepts multiple parameters suggestion with the `--search-space` option, the difference between the `optuna trial suggest` CLI and `Trial.suggest_*` APIs becomes larger. I prefer that the `optuna trial suggest` only accepts a single parameter with the `--distribution` option.

@toshihikoyanase Do you have any opinions on that?